dali/operators/random/batch_permutation.h

```diff
@@ -18,13 +18,14 @@
 #include <random>
 #include <vector>
 #include "dali/pipeline/operator/operator.h"
+#include "dali/operators/random/rng_base_cpu.h"
 
 namespace dali {
 
-class BatchPermutation : public Operator<CPUBackend> {
+class BatchPermutation : public rng::OperatorWithRng<CPUBackend, false> {
  public:
   explicit BatchPermutation(const OpSpec &spec)
-  : Operator<CPUBackend>(spec), rng_(spec.GetArgument<int64_t>("seed")) {}
+  : rng::OperatorWithRng<CPUBackend, false>(spec) {}
```

Comment:
Well..... I'm not a fan of this particular abuse ;) This operator only needs one RNG. Why bother creating (and serializing) all of them?